dashboard/app/dsl/script_dsl.rb

```diff
@@ -68,6 +70,7 @@ def level(name, properties = {})
     @levels << {
       :name => name,
       :stage => @stage,
+      :stage_flex_category => @stage_flex_category,
```

Comment:
The level doesn't need to know about the stage's `:flex_category`, but I agree this is super weird.  The reason it's written this way is due to how `add_script` works.

I'd expect `add_script` to add the script, then each stage, then each level in that stage.  But instead each `script_level` in the script is added, then the `script_level.stage` is found or created by name.  This is also the reason for `stage.save! if stage.changed?` below — only the first `script_level` in a stage actually needs `create!` the stage (or in this case, add the `:flex_category).